### PR TITLE
[READY TO MERGE AFTER TEST BY DANIEL][fixes] opds chain from main/convert to renderer/opds

### LINKED ISSUE #822: bug on the navigation pages on opds catalog
http://opoto-ed.org/catalogue/opds

When i click on the next button the url crash and do not go to the next page

image

last and first idem

### LINKED ISSUE #823: bug on getSearchUrlFromOpds1Feed
http://readium2.herokuapp.com/opds-v1-v2-convert/https%3A%2F%2Finstantclassics.librarysimplified.org%2Fchildrens-books.xml

debug:
```
readium-desktop:main/converter/opds getSearchUrlFromOpds1Feed TypeError: doc.Url.find is not a function
[2]     at OpdsFeedViewConverter.<anonymous> (/goinfre/thorium/dist/main.js:2658:53)
[2]     at Generator.next (<anonymous>)
[2]     at fulfilled (/goinfre/thorium/node_modules/tslib/tslib.js:107:62)
[2]     at processTicksAndRejections (internal/process/task_queues.js:89:5) +2s
```

Link:
```
{
            "href": "https://instantclassics.librarysimplified.org/search",
            "type": "application/opensearchdescription+xml",
            "rel": "search"
        },
```

search:
```
<?xml version="1.0" encoding="UTF-8"?>
 <OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
   <ShortName>Search</ShortName>
   <Description>Search</Description>
   <Tags></Tags>
   <Url type="application/atom+xml;profile=opds-catalog" template="https://instantclassics.librarysimplified.org/search?q={searchTerms}"/>
 </OpenSearchDescription>
```

discovered on #786 , with the fix of convertMultiLangStringToString

### PR DESCRIPTION
**Pull request by panaC**
- fixes #823 

- fixes #822

- fixes: GridView or ListView Store While Browsing the Catalog

- add: new header

- refactor converter opds, service opds, api opds, search, publicationInfo, publicationInfo saga, browser view, breadcrumb.

- moving the application code from the react view to redux saga side effect

- moving lcpInfo in publicationInfo to dedicated component

### CODE REVIEW

**Comment by panaC** [comment]
to test:

https://www.ebooksgratuits.com/opds/index.php
https://tools.wmflabs.org/wsexport/wikisource-fr-good.atom

**Comment by panaC** [comment]
image

http://readium2.herokuapp.com/opds-v1-v2-convert/https%3A%2F%2Fwww.ebooksgratuits.com%2Fopds%2Ffeed.php

**Comment by panaC** [comment]
opds1 to opds2 converter error : for opds.openedition.org
http://readium2.herokuapp.com/opds-v1-v2-convert/http%3A%2F%2Fopds.openedition.org%2F%3Fplatform%3DOJ%26core%3Dpublishers

image

should be navigation entry and not publication

**Comment by panaC** [comment]
full entry publication (rel: alternate) not detected on pUblicationInfo, why ?

http://readium2.herokuapp.com/opds-v1-v2-convert/https%3A%2F%2Fwww.atramenta.net%2Fopds%2Flatest.atom%3Ftype%3Dpublic_domain

**Comment by panaC** [comment]
with atramenta on public domains book. I can't download some epub.

http://readium2.herokuapp.com/opds-v1-v2-convert/https%3A%2F%2Fwww.atramenta.net%2Fopds%2Flatest.atom%3Ftype%3Dpublic_domain

https://www.atramenta.net/opds/download/78803.epub
https://www.atramenta.net/opds/download/77475.epub

returns "contactez-nous"

**Comment by danielweck** [comment]
> "contactez-nous"

`curl -s -L -I -X GET "https://www.atramenta.net/opds/download/78803.epub"`
HTTP 200 :(
```
HTTP/1.1 200 OK
Date: Wed, 27 Nov 2019 17:15:10 GMT
Server: Apache/2.4.29 (Ubuntu)
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
Set-Cookie: PHPSESSID=13eko8f9t4rr9f9vi1rs7v8566; path=/
Vary: Accept-Encoding
Transfer-Encoding: chunked
Content-Type: text/html; charset=ISO-8859-1
```

**Comment by danielweck** [comment]
Suggestion: filter-out / reject HTTP responses with `Content-Type` !== `application/epub+zip` (in this case: `text/html`).

**Comment by danielweck** [comment]
> should be navigation entry and not publication

Current heuristics:

https://github.com/readium/r2-opds-js/blob/6fb2d35e82a707f8257791532e4ae8bad640cbd2/src/opds/converter.ts#L314-L395

**Comment by danielweck** [comment]
Publication feed:
https://readium2.herokuapp.com/opds-v1-v2-convert/https%3A%2F%2Fcatalog.feedbooks.com%2Ffeatured%2Fen.atom
https://readium2.herokuapp.com/opds-v1-v2-convert/http%3A%2F%2Fmanybooks.net%2Fopds%2Fnew_titles.php
https://readium2.herokuapp.com/opds-v1-v2-convert/http%3A%2F%2Fmanybooks.net%2Fopds%2Fnew_titles.php
https://readium2.herokuapp.com/opds-v1-v2-convert/http%3A%2F%2Fsrv.manybooks.net%2Fopds%2Fsearch.php%3Fq%3DHello
https://readium2.herokuapp.com/opds-v1-v2-convert/https%3A%2F%2Fwww.ebooksgratuits.com%2Fopds%2Ffeed.php
https://readium2.herokuapp.com/opds-v1-v2-convert/https%3A%2F%2Fwww.atramenta.net%2Fopds%2Flatest.atom%3Ftype%3Dpublic_domain

Nvaigation feed:
https://readium2.herokuapp.com/opds-v1-v2-convert/http%3A%2F%2Fmanybooks.net%2Fopds%2Fgenres.php
https://readium2.herokuapp.com/opds-v1-v2-convert/http%3A%2F%2Fopds.openedition.org%2F%3Fplatform%3DOJ%26core%3Dpublishers

File: src/main/converter/opds.ts [hunk 1]
```diff
@@ -26,118 +25,211 @@ import { OPDSPublication } from "@r2-opds-js/opds/opds2/opds2-publication";
...
+debug("opds-converter");
+
+interface IGetLinksViewFilter {
+    rel?: string | string[] | RegExp;
+    type?: string | string[] | RegExp;
+}
+
+const supportedFileTypeLinkArray = [
+    "application/epub+zip",
+    "application/vnd.readium.lcp.license-1.0+json",
```
Comment (danielweck) [inline src/main/converter/opds.ts]: `application/vnd.readium.lcp.license-1.0+json`
should be:
`application/vnd.readium.lcp.license.1.0+json`
(`-` replaced with `.`)

Specification: https://readium.org/lcp-specs/releases/lcp/latest#32-content-conformance

Comment (panaC) [reply src/main/converter/opds.ts]: https://github.com/readium/readium-desktop/pull/832/commits/f692960474cbdc91b7677fe8950aecac7727f78c

**Comment by panaC** [comment]
It seems to me ready to merge this PR,
Thanks you Daniel,

**Comment by danielweck** [comment]
The build is borken in this PR (lint)
```
    ERROR in [at-loader] ./src/renderer/components/dialog/publicationInfos/TagManager.tsx:26:7 
        TS6133: 'debug' is declared but its value is never read.
```

**Comment by danielweck** [comment]
REGRESSION: this PR breaks LCP support in the publication info dialog (return / renew links)
Investigating ...

**Pull request merged**